backend/oidc/auth.py

```diff
@@ -0,0 +1,133 @@
+import logging
+from datetime import timedelta
+
+import requests
+from django.core.exceptions import SuspiciousOperation
+from django.urls import reverse
+from django.utils import timezone
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from mozilla_django_oidc.utils import absolutify
+
+from oidc.services import update_or_create_oidc_profile
+
+LOGGER = logging.getLogger(__name__)
+
+
+class HelsinkiOIDCAuthenticationBackend(OIDCAuthenticationBackend):
+    """Override Mozilla Django OIDC authentication."""
+
+    def verify_claims(self, claims):
+        """Override the original verify_claims method because it verifies the claim from the email and
+        email is not a mandatory field for suomi.fi authentication."""
+        return True
+
+    def filter_users_by_claims(self, claims):
+        """Return all users matching the specified username (sub)."""
+        username = claims.get("sub")
+        if not username:
+            return self.UserModel.objects.none()
+        return self.UserModel.objects.filter(username__iexact=username)
+
+    def create_user(self, claims):
+        """Return object for a newly created user account. Takte the values from the userinfo fields"""
+        return self.UserModel.objects.create_user(
+            username=claims.get("sub"),
+            first_name=claims.get("given_name"),
+            last_name=claims.get("family_name"),
+            email=claims.get("email"),
+        )
+
+    def store_token_info(self, user, token_info):
```

Comment:
I have a feeling that this function shouldn't belong to the `HelsinkiOIDCAuthenticationBackend` since it looks like a utility function. Would it make more sense if this can be moved to User model or the shared `service` ( just like the `update_or_create_oidc_profile`)

Reply:
Yep, that makes sense. I'll move it to `services.py`